metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -561,6 +562,51 @@ class Compilers(
 
   }
 
+  def syntheticDecorations(
+      path: AbsolutePath,
+      token: CancelToken,
+  ): Future[ju.List[DecorationOptions]] = {
+    loadCompiler(path)
+      .map { compiler =>
+        val (input, _, adjust) =
+          sourceAdjustments(
+            path.toNIO.toUri().toString(),
+            compiler.scalaVersion(),
+          )
+        val vFile =
+          CompilerVirtualFileParams(path.toNIO.toUri(), input.text, token)
+
+        val pcParams = CompilerSyntheticDecorationsParams(
+          vFile,
+          typeParameters = userConfig().showInferredType.contains("true"),
+          inferredTypes = userConfig().showInferredType.contains("minimal") ||
+            userConfig().showInferredType.contains("true"),
+          implicitParameters = userConfig().showImplicitArguments,
+          implicitConversions = userConfig().showImplicitConversionsAndClasses,
+        )
+        compiler
+          .syntheticDecorations(pcParams)
+          .asScala
+          .map { decorations =>
+            val decorationOptions = decorations.map { decoration =>
+              DecorationOptions(
+                decoration.label(),
+                adjust.adjustRange(decoration.range()),
+              )
+            }
+            val isScala2 = compiler.scalaVersion().startsWith("2.")
+            if (isScala2 && path.isWorksheet && !decorationOptions.isEmpty()) {
+              // In scala 2 worksheet, last decoration is for synthetic `main` method
+              decorationOptions.remove(decorationOptions.size() - 1)
```

Comment:
Should we generalize it and if something is outside the range of the document the it should not be pritned?

Reply:
Actually those symbols are in range so it's not easy to see if should not be printed. I added one more check in `PcSyntheticDecorationProvider.InferredType` that removes this `if` here, it also removes invalid decoration for `.sbt` I haven't noticed before.